llvm/include/llvm/IR/IntrinsicsRISCV.td

```diff
@@ -1705,19 +1705,39 @@ let TargetPrefix = "riscv" in {
 
   // Segment loads/stores for fixed vectors.
   foreach nf = [2, 3, 4, 5, 6, 7, 8] in {
+    // Input: (pointer, vl)
     def int_riscv_seg # nf # _load
           : DefaultAttrsIntrinsic<!listconcat([llvm_anyvector_ty],
                                               !listsplat(LLVMMatchType<0>,
                                               !add(nf, -1))),
                                   [llvm_anyptr_ty, llvm_anyint_ty],
                                   [NoCapture<ArgIndex<0>>, IntrReadMem]>;
+    // Input: (pointer, mask, vl)
```

Comment:
After this patch does anywhere still create the non-masked fixed vector intrinsics? If so could we convert them to the masked version and rely on RISCVVectorPeephole to unmask them?

Reply:
> After this patch does anywhere still create the non-masked fixed vector intrinsics?

The existing `TLI::lowerInterleavedLoad` and `TLI::lowerDeinterleavedIntrinsicToLoad` still use them. But I agree they can and should be replaced by the masked variant. I'll do it in a follow-up patch.

Reply:
It's fixed in #137045